### DISCUSSION TITLE
How to reuse the existing data directory?

### DISCUSSION THREAD

**Discussion by coordinatio** [LinwoodDev/Butterfly#751] [Q&A]
Hi everyone!

I've just switched from the F-Droid version to the version installed directly from the GitHub. I had to reinstall the app to perform the switch. Now I can not find a way to reimport all the data from the data directory into the app: I've pointed the app to the existing data directory and afterward it shows only the folders from there and no notes.

Is there a way to reimport all the notes and templates from the data folder, except by doing it one-by-one?

**Comment by CodeDoctorDE** [comment]
This is sadly a problem with the permission system of android. The app can only read files it has created.
You could try making a zip directory out of it and import this or just importing it on by one.

Im not a pro in android, if I forget a permission or there is a different way to access the files, feel free to correct me

(Im also suggesting you to use the fdroid version since it gets autoupdates)

**Comment by coordinatio** [comment]
Thanks for the quick response!

> You could try making a zip directory out of it and import this or just importing it on by one.

I've just tried that: Butterfly indeed have imported all the folders and the notes, but the notes are empty inside. Just white pages with no data.

Is there something I can do to try to fix that?

> (Im also suggesting you to use the fdroid version since it gets autoupdates)

Currently, F-Droid has 2.2.0 version, but the most recent is 2.2.1 with data directory related fixes. That made me switch from the F-Droid version.

**Reply by CodeDoctorDE** [reply]
Oh that looks like a bug. I will try to test this if im home.
Yeah, fdroid takes a time updating the packages

**Reply by coordinatio** [reply]
Tell me if you need help in reproducing the issue.

**Reply by CodeDoctorDE** [reply]
fixed, thanks for reporting, will be released in the next nightly next monday
